src/target/riscv/riscv-013.c

```diff
@@ -5076,8 +5078,10 @@ static int riscv013_write_debug_buffer(struct target *target, unsigned int index
 static riscv_insn_t riscv013_read_debug_buffer(struct target *target, unsigned int index)
 {
 	uint32_t value;
-	dm_read(target, &value, DM_PROGBUF0 + index);
-	return value;
+	if (dm_read(target, &value, DM_PROGBUF0 + index) == ERROR_OK)
+		return value;
+	else
+		return 0;
```

Comment:
I have the same concern as with `read_abstract_arg()`. Also, wouldn't initializing the `value` to zero fix the warning?

Reply:
Yes, initializing to 0 would fix the warning, but I think this makes it more clear what is going on. It also highlights that we're ignoring the error better.